### PR TITLE
Claim relayer will send tx to CENNZnet listen to deposit on ethereum …

### PR DESCRIPTION
**Pull request by KarishmaBothara**
This PR provides

1. claimer service to listen to deposit on Ethereum bridge contract and deposit the claim after waiting for confirmation on CENNZnet side
2. Also stores the claim info in mongo db
3. User can check if the status of their tx using
```
http://localhost:3000/bridge/txHash/0x42dd98665329917a0aecca588faedfabe2d367c5d46e78b4c4019dee19b77724
```
status is either - waiting for confirmation, submitted, failed or successful

### CODE REVIEW

File: scripts/subscribeEthereumDeposit.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                }
+            }
+        });
+    });
+}
+
+async function main (networkName, pegContractAddress) {
+    networkName = networkName || 'local';
+
+    const connectionStr = `mongodb://${process.env.MONGO_INITDB_ROOT_USERNAME}:${process.env.MONGO_INITDB_ROOT_PASSWORD}@${process.env.MONGO_HOST}:${process.env.MONGO_PORT}/admin`;
```
Comment (holygits) [inline scripts/subscribeEthereumDeposit.js:55]:
```suggestion
    const connectionStr = `mongodb://${process.env.MONGO_INITDB_ROOT_USERNAME}:${process.env.MONGO_INITDB_ROOT_PASSWORD}@${process.env.MONGO_HOST}:${process.env.MONGO_PORT}/${process.env.MONGO_COLLECTION_NAME}`;
```

**Review by holygits — APPROVED** [review]
lgtm - few nits

File: scripts/subscribeEthereumDeposit.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                }
+            }
+        });
+    });
+}
+
+async function main (networkName, pegContractAddress) {
+    networkName = networkName || 'local';
+
+    const connectionStr = `mongodb://${process.env.MONGO_INITDB_ROOT_USERNAME}:${process.env.MONGO_INITDB_ROOT_PASSWORD}@${process.env.MONGO_HOST}:${process.env.MONGO_PORT}/${process.env.MONGO_COLLECTION_NAME}`;
```
Comment (holygits) [inline scripts/subscribeEthereumDeposit.js:55]: sry should be:
```suggestion
    const connectionStr = `mongodb://${process.env.MONGO_INITDB_ROOT_USERNAME}:${process.env.MONGO_INITDB_ROOT_PASSWORD}@${process.env.MONGO_HOST}:${process.env.MONGO_PORT}/${process.env.MONGO_DB_NAME}`;
```

File: src/mongo/models.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const mongoose = require('mongoose');
+const { Schema } = mongoose;
+
+
+const BRIDGE_CLAIM = 'bridgeClaim';
```
Comment (holygits) [inline src/mongo/models.js:5]: last one. shorter and we know it's related to bridge already:
```suggestion
const BRIDGE_CLAIM = 'claims';
```

File: .github/workflows/image-builder.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      - "**.lock"
+      - "genesis/**"
+      - "Dockerfile"
+
+jobs:
+  build-image:
+    # only run if prechecks pass and the source branch is named 'prerelease/*'
+    runs-on: [self-hosted, linux]
+    env:
+      IMAGE_NAME: "cennznet/depositClaim"
```
Comment (holygits) [inline .github/workflows/image-builder.yml:27]: convention for image names it `-`
```suggestion
      IMAGE_NAME: "cennznet/bridge-relayer-service"
```

File: .github/workflows/image-builder.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
+# Build a docker image on push to main
+# Also tag the image if commit is tagged
+
+name: "Image Builder"
+
+on:
+  release:
+    types:
+      - "created"
+      - "released"
```
Comment (holygits) [inline .github/workflows/image-builder.yml:10]: don't need this part

File: .github/workflows/image-builder.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  push:
+    branches:
+      - "main"
+    paths:
+      - "**.rs"
+      - "**.toml"
+      - "**.yml"
+      - "**.lock"
+      - "genesis/**"
+      - "Dockerfile"
```
Comment (holygits) [inline .github/workflows/image-builder.yml:20]:
```suggestion
      - "**.js"
      - "**.json"
      - "Dockerfile"
```

File: scripts/subscribeEthereumDeposit.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+const { Api } = require('@cennznet/api');
+require("dotenv").config();
+const { Keyring } = require('@polkadot/keyring');
+const logger = require('./logger');
+const mongoose = require('mongoose');
+const { BridgeClaim  } = require('../src/mongo/models');
+const { ethers } = require("hardhat");
+let txExecutor;
+
+const airDropAmount = 5000;
```
Comment (holygits) [inline scripts/subscribeEthereumDeposit.js:11]:
```suggestion
const airDropAmount = 50000;
```

File: scripts/subscribeEthereumDeposit.js [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    const peg = await Peg.attach(pegContractAddress);
+    await peg.deployed();
+    logger.info(`CENNZnet peg deployed to: ${peg.address}`);
+    logger.info(`Executor: ${txExecutor.address}`);
+    const eventConfirmation = (await api.query.ethBridge.eventConfirmations()).toNumber();
+    console.log('eventConfirmation::',eventConfirmation);
+    peg.on("Deposit", async (sender, tokenAddress, amount, cennznetAddress, eventInfo) => {
+        let nonce = (await api.rpc.system.accountNextIndex(claimer.address)).toNumber();
+        console.log('Nonce:::', nonce);
+        const checkIfBridgePause = await api.query.ethBridge.bridgePaused();
```
Comment (holygits) [inline scripts/subscribeEthereumDeposit.js:103]: would be good if we stored these and resubmitted them when bridge is unpaused again.
Keen to merge this as is however

**Review by holygits — APPROVED** [review]

File: scripts/subscribeEthereumDeposit.js [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+const airDropAmount = 50000;
+
+async function airDrop(claimId, signer, api, spendingAssetId) {
+    const signerBalance = await api.query.genericAsset.freeBalance(spendingAssetId, signer.address);
+    if (signerBalance.toNumber() > airDropAmount) {
+        const record = await BridgeClaim.findOne({claimId});
+        const cennznetAddress = record.cennznetAddress;
+        logger.info(`Air drop in progress for address ${cennznetAddress}`);
+        await api.tx.genericAsset.transfer(spendingAssetId, cennznetAddress, airDropAmount).signAndSend(signer);
```
Comment (holygits) [inline scripts/subscribeEthereumDeposit.js:20]: cap airdrop to once per address max.
shouldn't be able to get tokens for multiple claims from same account

File: scripts/subscribeEthereumDeposit.js (hunk 8 above)
Comment (KarishmaBothara) [reply scripts/subscribeEthereumDeposit.js:103]: I was hoping when we pause from CENNZnet, we would pause on Ethereum side too

File: scripts/subscribeEthereumDeposit.js (hunk 9 above)
Comment (KarishmaBothara) [reply scripts/subscribeEthereumDeposit.js]: done.

File: scripts/subscribeEthereumDeposit.js (hunk 8 above)
Comment (holygits) [reply scripts/subscribeEthereumDeposit.js:106]: during the last session of an era the bridge is paused on CENNZnet side to ensure that a proof is generated for the next validator set.

Comment (KarishmaBothara) [reply scripts/subscribeEthereumDeposit.js:106]: oh ok

**Pull request merged**